EmployeeLib/Employee.cs

```diff
@@ -1,24 +1,34 @@
 ﻿namespace EmployeeLib
 {
-    public class Employee
+    public abstract class EmployeeBase
     {
         public string FirstName { get; set; }
         public string LastName { get; set; }
-        public Employee Manager { get; set; } = null;
         public decimal Salary { get; set; }
-
-        public virtual void AssignManager(Employee manager)
-        {
-            // Simulate doing other tasks here - otherwise, this should be
-            // a property set statement, not a method.
-            Manager = manager;
-        }
-
         public virtual void CalculatePerHourRate(int rank)
         {
             decimal baseAmount = 12.50M;
 
             Salary = baseAmount + (rank * 2);
         }
     }
+
+    public interface ISupervised
+    {
+        EmployeeBase Supervisor { get; set; }
+        void AssignManager(EmployeeBase manager);
+    }
+
+    public class Employee : EmployeeBase, ISupervised
+    {
+
+        public EmployeeBase Supervisor { get; set; }
+        public void AssignManager(EmployeeBase manager)
```

Comment:
assigning a manager should only take employees of type manager or sub types of manager